part3/identifying-high-leverage-activities.qmd

```diff
@@ -0,0 +1,284 @@
+
+Take insurance claims processing as an example. An insurance claim can go through multiple rounds of asking for evidence for a claim, verifying that evidence, and asking the customer for more information. 
+
+In another example, a telco customer support agent that investigates an unexpectedly high bill charge may need to first identify the cause of the anomaly, and then wait for a human to approve a refund. 
+
+To address this complexity, we've identified a repeatable sequence of planning activities that we've found to be important for mapping out our problem space and for aligning stakeholders. Focus on these key areas at the start of a project, and the project's chances of success increase greatly. 
+
+These activities can be implemented as part of a ~two week design and planning workshop at the start of a project, something like a design sprint. Design sprints are time-boxed collaborative workshops that bring together cross-functional teams to rapidly prototype and validate solutions before committing significant development resources. Originally popularized by Google Ventures, design sprints compress months of traditional planning into an intensive, focused period where teams can align on design decisions, identify key data sources and plan evaluation suites before diving into the complex technical implementation.
+
+## Key planning activities to prioritize
+
+To address this complexity, we've identified a repeatable sequence of planning activities that we've found to be important for mapping out our problem space and for aligning stakeholders. Focus on these key areas at the start of a project, and the project's chances of success increase greatly. 
+
+These activities can be implemented as part of a roughly two-week design and planning workshop at the start of a project, something like a design sprint. Design sprints are time-boxed collaborative workshops that bring together cross-functional teams to rapidly prototype and validate solutions before committing significant development resources. Originally popularized by Google Ventures, design sprints compress months of traditional planning into an intensive, focused period where teams can align on design decisions, identify key data sources and plan evaluation suites before diving into the complex technical implementation.
+
+### Suggested planning sequence
+
+1. Map out in detail the current workflow as it is executed by humans
+2. Translate the business workflow mapped in step 1 into a technical architecture covering:
+    - the required structured and unstructured data sources
+    - which aspects of the workflow should be handled by tool-calling sub-agents or by linear workflows 
+    - how each sub-component should be evaluated for quality  
+3. List out the information that different stakeholders, from C-level executives to business users to developers need from the agentic application
+4. Design tracing and logging modules addressing the needs listed in step 3. 
```

Comment:
What does "tracing & logging modules" mean specifically?

Reply:
In reality we'd cycle through steps 2-5 iteratively, but I usually find putting in some kind of traces as early as possible is helpful because it shows what the default `mlflow.autolog` doesn't cover and prompts you to think where custom spans need to be added. Maybe we could rephrase as "design first pass of tracing and logging modules, refining the tracing logic with future iterations"?